[PL-50980] PL 1.37.6 Release Notes (#6628)
harness/developer-hub: release-notes/platform.md

## May 2024

### Version 1.37.x<!--  May 13, 2024 -->

#### Fixed issues

> - An "uuid may not be empty" error occurred when configuring LDAP settings. This was due to the LDAP identifier being sent as an empty string instead of null if it didn't already exist. The issue has been resolved, and the identifier is now correctly sent as null, preventing the error. (PL-50657)

Reviewer: ```suggestion
- An `UUID may not be empty` error occurred when configuring LDAP settings. This was due to the LDAP identifier being sent as an empty string instead of null if it didn't already exist. The issue has been resolved, and the identifier is now correctly sent as null, preventing the error. (PL-50657)
```

> - The Cisco (CFD) getUserGroupList API was returning inconsistent results, such as missing groups, due to issues with the sorting functionality based solely on the `lastModifiedAt` timestamp. We've enhanced the sorting mechanism in the User Group List API by introducing a secondary sort field to serve as a tiebreaker. This ensures consistent and reliable ordering across queries, significantly improving the stability and accuracy of paginated results. (PL-48886, ZD-61135)

Reviewer: ```suggestion
- The Cisco (CFD) `getUserGroupList` API was returning inconsistent results, such as missing groups, due to issues with the sorting functionality based solely on the `lastModifiedAt` timestamp. We've enhanced the sorting mechanism in the User Group List API by introducing a secondary sort field to serve as a tiebreaker. This ensures consistent and reliable ordering across queries, significantly improving the stability and accuracy of paginated results. (PL-48886, ZD-61135)
```

> - The Cisco (CFD) `getUserGroupList` API was returning inconsistent results, such as missing groups, due to issues with the sorting functionality based solely on the `lastModifiedAt` timestamp. We've enhanced the sorting mechanism in the User Group List API by introducing a secondary sort field to serve as a tiebreaker. This ensures consistent and reliable ordering across queries, significantly improving the stability and accuracy of paginated results. (PL-48886, ZD-61135)

Reviewer 2: Please don't have customer name in the release notes.

> - The Cisco (CFD) `getUserGroupList` API was returning inconsistent results, such as missing groups, due to issues with the sorting functionality based solely on the `lastModifiedAt` timestamp. We've enhanced the sorting mechanism in the User Group List API by introducing a secondary sort field to serve as a tiebreaker. This ensures consistent and reliable ordering across queries, significantly improving the stability and accuracy of paginated results. (PL-48886, ZD-61135)

Reviewer: ```suggestion
- The `getUserGroupList` API was returning inconsistent results, such as missing groups, due to issues with the sorting functionality based solely on the `lastModifiedAt` timestamp. We've enhanced the sorting mechanism in the User Group List API by introducing a secondary sort field to serve as a tiebreaker. This ensures consistent and reliable ordering across queries, significantly improving the stability and accuracy of paginated results. (PL-48886, ZD-61135)
```

> - Login issues occurred in NG when FG delegates were scaled down because the LDAP Authentication task was initially sent to CG Delegates. If no CG Delegate was available, the task would expire before being sent to an NG Delegate. Compounding the issue, the timer at the Gateway was set to expire before the delegate task, leading to failed login attempts. We have implemented a solution that prioritizes sending the LDAP Authentication task to NG Delegates first if the feature flag `PL_USE_NG_DELEGATE_LDAP_AUTH` is enabled, thereby enhancing the reliability of login processes in NG environments. (PL-48541, ZD-60437)

Reviewer: ```suggestion
- Login issues occurred in NextGen when FirstGen delegates were scaled down because the LDAP Authentication task was initially sent to FirstGen Delegates. If a FirstGen delegate wasn't available, the task would expire before being sent to a NextGen delegate. Compounding the issue, the timer at the Gateway was set to expire before the delegate task, leading to failed login attempts. We have implemented a solution that prioritizes sending the LDAP Authentication task to NextGen delegates first if the feature flag `PL_USE_NG_DELEGATE_LDAP_AUTH` is enabled, thereby enhancing the reliability of login processes in NextGen environments. (PL-48541, ZD-60437)
```

> - The AWS connector was failing due to permission issues related to the AWS region in use. Specifically, the aws-java-sdk-core version 1.12.261 does not support the ap-south-2 region, resulting in an UnrecognizedClientException. The support to add new regions is taken up as an enahncement requiring the AWS SDK upgrade and changing Harness UI and backend flows. (PL-48451, ZD-58686)

Reviewer: ```suggestion
- The AWS connector was failing due to permission issues related to the AWS region in use. Specifically, the `aws-java-sdk-core` version 1.12.261 does not support the ap-south-2 region, resulting in an unrecognized client exception. The support to add new regions is taken up as an enhancement requiring the AWS SDK upgrade and changing Harness UI and backend flows. (PL-48451, ZD-58686)
```

> Customers encountered problems with Pod Disruption Budgets (PDBs) in the Harness namespace, leading to issues during patching cycles due to incorrectly targeted PDB rules. We have updated the PDB rules for the `cloud-info`, `anomaly-detection`, `gateway`, and `next-gen-ui` services to accurately select the correct pods, ensuring smooth operations during maintenance activities. (PL-49350, ZD-62353)

Reviewer: ```suggestion
- Problems with Pod Disruption Budgets (PDBs) in the Harness namespace caused issues during patching cycles due to incorrectly targeted PDB rules. Harness updated PDB rules for the `cloud-info`, `anomaly-detection`, `gateway`, and `next-gen-ui` services to select the correct pods, ensuring smooth operations during maintenance activities. (PL-49350, ZD-62353)
```
